src/test/java/com/myworkspace/reportapp/entity/customer/CustomerTest.java

```diff
@@ -0,0 +1,49 @@
+package com.myworkspace.reportapp.entity.customer;
+
+import com.myworkspace.reportapp.entity.EntityTest;
+import com.myworkspace.reportapp.entity.device.Device;
+import com.myworkspace.reportapp.entity.device.DeviceBase;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CustomerTest extends EntityTest {
+
+    @Test
+    void shouldSaveCustomer() {
+        //GIVEN
+        final var customer = new Customer("gmail@wp.pl",
+                "669876222",
+                "IT AG",
+                "Jerozolimskie 100, 02-017, Warsaw, Poland");
+        //WHEN
+        persistObject(customer);
+        //THEN
+        final var readCustomer = em.find(Customer.class, customer.getId());
+        assertEquals(customer, readCustomer);
+
+    }
+
+    @Test
+    void shouldSaveCustomerWithDevice() {
```

Comment:
It's generally a good practice to have each test focus on a specific scenario or aspect. Consider breaking down the method into two separate tests: one to test saving a customer and another to test associating a device with a customer. This promotes better isolation and clearer identification of issues if they arise.

Reply:
Add device into DeviceTest instead of CustomerTest.